packages/transaction-controller/src/TransactionController.ts

```diff
@@ -443,6 +447,11 @@ export class TransactionController extends BaseController<
       origin,
     );
 
+    const { type: transactionType } = await determineTransactionType(
```

Comment:
This is much cleaner, but does this mean we're always doing additional callouts and logic even when perhaps not required?

Could we just fallback here like:

`const transactionType = type ?? (await determineTransactionType(txParams, this.ethQuery)).type;`

Reply:
100%, just pushed the change you suggested. :pray: